src/node_zlib.cc

```diff
@@ -254,11 +253,22 @@ class ZCtx : public AsyncWrap {
             ctx->err_ = Z_NEED_DICT;
           }
         }
+        if (ctx->strm_.avail_in >= 2 && ctx->mode_ == GUNZIP) {
+          // Bytes remain in input buffer. Perhaps this is another compressed
+          // member in the same archive, or just trailing garbage.
+          // Check the header to find out.
+          if (*reinterpret_cast<uint16_t*>(ctx->strm_.next_in)
+              != GZIP_HEADER_ID) {
```

Comment:
Don't do this; unaligned loads can SIGBUS on some architectures (notably ARM and MIPS.)